### PR TITLE
chore: update macos version to latest

### PR DESCRIPTION
**Pull request by Samaritan1011001**
# DO NOT MERGE, prereq: https://github.com/aws-amplify/amplify-js-samples-staging/pull/859

#### Description of changes

Reverts the macos version to latest. This needs to be merged after the [staging repo PR](https://github.com/aws-amplify/amplify-js-samples-staging/pull/859) so the detox upgrade is in.

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/push-integ-test.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ concurrency:
 on:
   push:
     branches:
-      - next/main
+      - manoj-update-detox-version
```
Comment (Samaritan1011001) [inline .github/workflows/push-integ-test.yml:11]: will revert when the staging PR is merged in.

Comment (jimblanc) [reply .github/workflows/push-integ-test.yml:11]: TY, assume we had a clean test run?

**Review by jimblanc — COMMENTED** [review]
LGTM, will approve once testing change reverted

File: .github/workflows/push-integ-test.yml (hunk 1 above)
Comment (ashika112) [reply .github/workflows/push-integ-test.yml:11]: Yes :) https://github.com/aws-amplify/amplify-js/actions/runs/8853977044

Comment (ashika112) [reply .github/workflows/push-integ-test.yml:11]: I think its ok to revert this and get approvals and merge both together

Comment (Samaritan1011001) [reply .github/workflows/push-integ-test.yml]: reverted it

File: .github/integ-config/integ-all.yml [hunk 2]
```diff
@@ -16,822 +16,737 @@ tests:
...
-  - test_name: integ_react_datastore_consecutive_saves_v2
-    desc: 'DataStore - Subs Disabled CLI v2'
-    framework: react
-    category: datastore
-    sample_name: [v2/consecutive-saves-v2]
-    spec: consecutive-saves
-    browser: *minimal_browser_list
-  # - test_name: integ_react_datastore_observe_query
-  #   desc: 'DataStore - Observe Query'
+  # - test_name: integ_datastore_auth-static-user-pool-groups-default
```
Comment (ashika112) [inline .github/integ-config/integ-all.yml:19]: this needs to be reverted as well

**Review by ashika112 — APPROVED** [review]

**Review by ashwinkumar6 — APPROVED** [review]

**Review by jimblanc — APPROVED** [review]

**Pull request merged**